File: 프로그래머스/unrated/178871. 달리기 경주/달리기 경주.py
def solution(players, callings):
    player_dict = {player: idx for idx, player in enumerate(players)}
    for calling in callings:
        current_idx = player_dict[calling]
        # print(current_idx)
        player_dict[calling] -= 1
        # print(player_dict[calling])
        player_dict[players[current_idx-1]] += 1
        # print(player_dict[players[current_idx-1]])
        players[current_idx-1], players[current_idx] = players[current_idx], players[current_idx-1]
        
    return players

File: 프로그래머스/1/178871. 달리기 경주/달리기 경주.js
function solution(players, callings) {
    const players_dict = {};

    players.forEach((player, idx) => {
        players_dict[player] = idx;
    });

    callings.forEach((calling) => {
        const currentIdx = players_dict[calling];
        players_dict[calling] -= 1;
        players_dict[players[currentIdx-1]] += 1;
        [players[currentIdx-1], players[currentIdx]] = [players[currentIdx], players[currentIdx-1]];
    });

    return players;
};
